### PR TITLE
Feature/support different surveys

### PR DESCRIPTION
**Pull request by AndrewTorrance**
You should also be aware that some of the tests added give false positives when run via pycharm which looks like a bug in pycharm with multiple patches. The tests are reliable when run under make 

## Checklist
  - CHANGELOG.md updated

### CODE REVIEW

File: app/tests/test_consumer.py [hunk 1]
```diff
@@ -12,10 +12,11 @@
 from sdc.crypto.key_store import KeyStore
 from sdc.crypto.exceptions import InvalidTokenException
 from sdc.rabbit.exceptions import QuarantinableError, RetryableError
+from unittest.mock import patch
 import yaml
 
 from app.main import SeftConsumer, KEY_PURPOSE_CONSUMER
-from app.settings import RM_SDX_GATEWAY_URL
+from app.settings import RM_SDX_GATEWAY_URL  # FTP_FOLDER_BASE
```
Comment (warrenbailey) [inline app/tests/test_consumer.py]: Don't think this needs to be here:  # FTP_FOLDER_BASE

File: README.md [hunk 2]
```diff
@@ -47,7 +47,8 @@ The main configuration options are listed below:
 | SEFT_FTP_PORT                         | `2021`                         | FTP port
 | SEFT_FTP_USER                         | `ons`                          | FTP username
 | SEFT_FTP_PASS                         | `ons`                          | FTP password
-| SEFT_CONSUMER_FTP_FOLDER              | `.`                            | FTP Folder
+| SEFT_CONSUMER_FTP_FOLDER              | `.`                            | FTP Folder (Deprecated)
```
Comment (warrenbailey) [inline README.md]: Lets just remove it rather than deprecated it

We'll release v2.0.0 when this is merged instead

File: app/settings.py [hunk 3]
```diff
@@ -25,7 +25,10 @@
 FTP_PORT = int(os.getenv('SEFT_FTP_PORT', '2021'))
 FTP_USER = os.getenv('SEFT_FTP_USER', 'ons')
 FTP_PASS = os.getenv('SEFT_FTP_PASS', 'ons')
-FTP_FOLDER = os.getenv('SEFT_CONSUMER_FTP_FOLDER', '.')
+
+FTP_FOLDER = os.getenv('SEFT_CONSUMER_FTP_FOLDER', '.')     # Deprecated with the introduction of FTP_FOLDER__BASE
```
Comment (warrenbailey) [inline app/settings.py]: As above

File: app/tests/test_consumer.py (hunk 1 above)
Comment (AndrewTorrance) [reply app/tests/test_consumer.py]: Changed

File: app/settings.py (hunk 3 above)
Comment (AndrewTorrance) [reply app/settings.py]: Changed

File: README.md (hunk 2 above)
Comment (AndrewTorrance) [reply README.md]: Changed

**Comment by AndrewTorrance** [comment]
[Update SDX seft consumer service to use survey id to decide FTP directory (3)](https://trello.com/c/Fu4KYmh6/562-update-sdx-seft-consumer-service-to-use-survey-id-to-decide-ftp-directory-3)

File: app/tests/test_consumer.py [hunk 4]
```diff
@@ -30,20 +31,76 @@ def setUp(self):
...
+            payload = '{"filename":"test1.xls", "file":"' + encoded_contents.decode() + \
+                      '", "case_id": "601c4ee4-83ed-11e7-bb31-be2e44b06b34", "survey_id": "SomeSurveyId"}'
+            payload_as_json = json.loads(payload)
+            encrypted_jwt = encrypt(payload_as_json, self.ras_key_store, KEY_PURPOSE_CONSUMER)
+            self.consumer.process(encrypted_jwt, tx_id)
+
+        mock_send_receipt.assert_called_with('601c4ee4-83ed-11e7-bb31-be2e44b06b34', tx_id)
+
+    @patch('app.main.SeftConsumer._send_receipt')
+    @patch.object(SDXFTP, 'deliver_binary', )
```
Comment (insacuri) [inline app/tests/test_consumer.py]: Is the trailing comma intentional?

Comment (AndrewTorrance) [reply app/tests/test_consumer.py]: nope , thanks , changed

File: app/main.py [hunk 5]
```diff
@@ -88,20 +91,21 @@ def process(self, encrypted_jwt, tx_id=None):
             decrypted_payload = self._decrypt(encrypted_jwt, tx_id)
 
             bound_logger.info("Extracting file")
-            decoded_contents, file_name, case_id = self.extract_file(decrypted_payload, tx_id)
+            decoded_contents, file_name, case_id, survey_id = self.extract_file(decrypted_payload, tx_id)
```
Comment (JamesGardiner) [inline app/main.py]: returning this many variables from a method could be made more explicit and cleaner by using a Named Tuple.

File: app/main.py [hunk 6]
```diff
@@ -156,6 +160,11 @@ def stop(self):
         logger.debug("Stopping consumer")
         self.consumer.stop()
 
+    @staticmethod
+    def _get_ftp_file_path(survey_id: str):
```
Comment (JamesGardiner) [inline app/main.py]: Will this even work with Python 3.4?

File: app/sdxftp.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import io
 from ftplib import FTP
+from os.path import join
```
Comment (JamesGardiner) [inline app/sdxftp.py]: Alphabetized imports

File: app/tests/test_consumer.py [hunk 8]
```diff
@@ -12,6 +12,7 @@
 from sdc.crypto.key_store import KeyStore
 from sdc.crypto.exceptions import InvalidTokenException
 from sdc.rabbit.exceptions import QuarantinableError, RetryableError
+from unittest.mock import patch
```
Comment (JamesGardiner) [inline app/tests/test_consumer.py]: PEP8 compliant imports please

File: app/main.py [hunk 9]
```diff
@@ -44,14 +44,17 @@ def extract_file(decrypted_payload, tx_id):
             file_contents = decrypted_payload['file']
             file_name = decrypted_payload['filename']
             case_id = decrypted_payload['case_id']
-            if not file_name or not file_contents or not case_id:
+            survey_id = decrypted_payload['survey_id']
```
[the remark this answers is not in the document]
Comment (AndrewTorrance) [reply app/main.py]: These checks are specifically to raise an error under these conditions , not to act as a filter to other statements . Hence I think it should remain like this

**Comment by warrenbailey** [comment]
Hold off merging this until after the deployment today

**Pull request merged**